frontend/src/components/studySamples/StudyStepSamplesTable.tsx

```diff
@@ -0,0 +1,172 @@
+import { Table, TableColumnType } from 'antd'
+import type { ColumnsType } from 'antd/es/table'
+import React, { useEffect, useState } from 'react'
+import { Link } from 'react-router-dom'
+import { useAppSelector } from '../../hooks'
+import { ItemsByID, Library, Protocol, Sample } from '../../models/frontend_models'
+import { StudySampleStep } from '../../models/study_samples'
+import { selectLibrariesByID, selectProtocolsByID, selectSamplesByID } from '../../selectors'
+import { withSample } from '../../utils/withItem'
+import { ProtocolNames } from '../../models/protocols'
+
+
+import SAMPLE_COLUMNS, { ObjectWithSample } from './SampleTableColumns'
+import LIBRARY_COLUMNS, { ObjectWithLibrary } from './LibraryTableColumns'
+
+
+
+
+interface SampleAndLibrary extends ObjectWithSample, ObjectWithLibrary {}
+
+function getColumnsForStep(step: StudySampleStep, protocol: Protocol): ColumnsType<SampleAndLibrary> {
+
+	const DEFAULT_SAMPLE_COLUMNS = [
```

Comment:
The idea of default sample and library is not bad ... There is a problem with QC though since the listed sample will not have gone through QC yet... This is the columns i would add.
1. default sample
id
kind
name
container_barcode
coordinates
volume
conc.
qc_flag
creation_date
depleted

2. default library
id
kind
name
container_barcode
coordinates
index
volume
conc.
qc_flag
creation_date
depleted

3. experiment
id
name
container_barcode
coordinates
volume
# samples => nice to have but not required.
qc_flag
creation_date
depleted

Not sure we need anything else. Even then we could strip a couple columns most likely, the lab work page is where people will check for these informations.